firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java

```diff
@@ -132,4 +140,55 @@ private PublicKey parseKey(String base64PublicKey) {
     }
     return null;
   }
+
+  private static String getDefaultSenderId(FirebaseApp app) {
+    // Check for an explicit sender id
+    String senderId = app.getOptions().getGcmSenderId();
+    if (senderId != null) {
+      return senderId;
+    }
+    String appId = app.getOptions().getApplicationId();
+    if (!appId.startsWith("1:")) {
+      // Not v1, server should be updated to accept the full app ID now
+      return appId;
+    } else {
+      // For v1 app IDs, fall back to parsing the project ID out
```

Comment:
a) I don't follow: v2 App-IDs also contain the Project-Number that can be parsed.
b) Do you mean Project-ID or Project-Number here?